napari/layers/labels/labels.py

```diff
@@ -476,11 +476,33 @@ def show_selected_label(self, filter):
     @property
     def brush_shape(self):
         """str: Paintbrush shape"""
+
+        warnings.warn(
+            (
+                "The square brush shape is deprecated and will be removed in version 0.4.9. "
+                "Afterward, only the circle brush shape will be available, "
+                "and the layer.brush_shape attribute will be removed."
+            ),
+            category=FutureWarning,
+            stacklevel=2,
+        )
+
         return str(self._brush_shape)
 
     @brush_shape.setter
     def brush_shape(self, brush_shape):
         """Set current brush shape."""
+
+        warnings.warn(
+            (
+                "The square brush shape is deprecated and will be removed in version 0.4.9. "
+                "Afterward, only the circle brush shape will be available, "
+                "and the layer.brush_shape attribute will be removed."
+            ),
+            category=FutureWarning,
+            stacklevel=2,
+        )
```

Comment:
:+1: thanks for making long and explicit deprecation warnings with versions number.

For my own understanding: Any policies on FutureWarning vs (Pending)DeprecationWarning ?

Reply:
@sofroniewn What are your thoughts on the warning?  I just used the future warning because that was in the example you pointed to, but I am not sure if its what should be used here.

also, @goanpeca will the translation work on the warning?  should I change this to be wrapped in `trans_()`?

Reply:
Please do :)

Reply:
> For my own understanding: Any policies on FutureWarning vs (Pending)DeprecationWarning ?

I think we did this once a while back and maybe I'm just copying and pasting every time. I'd be happy to defer to an alternative if its the recommended best practice, I don't have strong opinions here. Maybe we open up a new issue to discuss though, and if we decide to change quickly change all of them.

Reply:
DeprecationWarnings are not visible by default:

https://docs.python.org/3/library/exceptions.html#DeprecationWarning

We want these warnings to be visible to users without having to enable Python development mode.

Reply:
they can potentially be made visible in, for instance, IPython or a script, as long as you adjust `stacklevel` appropriately...  Just in case that puts it back on the table (to use the warning that feels semantically more appropriate)?

Reply:
so... with a `warn(..., DeprecationWarning, stacklevel=2)`, if someone directly accessed `labels_layer.brush_shape` the warning _would_ be visible to the user without dev mode.

Reply:
Makes sense, lets do in a follow up though

Reply:
@sofroniewn @tlambert03 the issue though is that many users will not be setting these things directly in `__main__` but in functions/methods deep in their libraries. @tlambert03 points out that pytest sets dev mode and so you'd see them there, but not everyone has perfect test coverage...